services/catalog/src/simcore_service_catalog/services/rabbitmq.py

```diff
@@ -15,25 +17,21 @@ def get_rabbitmq_settings(app: FastAPI) -> RabbitSettings:
     return settings
 
 
-def setup_rabbitmq(app: FastAPI) -> None:
+async def setup_rabbitmq(app: FastAPI) -> AsyncIterator[State]:
     settings: RabbitSettings = get_rabbitmq_settings(app)
-    app.state.rabbitmq_rpc_server = None
+    await wait_till_rabbitmq_responsive(settings.dsn)
 
-    async def _on_startup() -> None:
-        await wait_till_rabbitmq_responsive(settings.dsn)
+    app.state.rabbitmq_rpc_server = await RabbitMQRPCClient.create(
+        client_name=f"{PROJECT_NAME}_rpc_server", settings=settings
+    )
 
-        app.state.rabbitmq_rpc_server = await RabbitMQRPCClient.create(
-            client_name=f"{PROJECT_NAME}_rpc_server", settings=settings
-        )
-
-    async def _on_shutdown() -> None:
+    try:
+        yield {}
```

Comment:
one question: I thought that it would automatically store the stuff in the app.state. why not here?

Reply:
@sanderegg There are two states. One for the manager (that you yield here or can get in the input) and another for the `app`  (i.e. `app.state`). See this example https://github.com/ITISFoundation/osparc-simcore/blob/a9ecdced2e41be04eff7252625155a17299bc025/packages/service-library/tests/fastapi/test_postgres_lifespan.py#L97-L108


The idea is that every lifespan function can get and/or update the manager's state.  The `app.state` is something separated that can be indeed modified in these functions as well.